crates/libs/core/src/types/client/mod.rs

```diff
@@ -5,8 +5,50 @@
 
 // This mod contains fabric client related types
 mod partition;
+use mssf_com::FabricTypes::{
+    FABRIC_SERVICE_NOTIFICATION_FILTER_DESCRIPTION, FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS,
+    FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS_NAME_PREFIX,
+    FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS_NONE,
+    FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS_PRIMARY_ONLY, FABRIC_URI,
+};
 pub use partition::*;
 mod node;
 pub use node::*;
 mod replica;
 pub use replica::*;
+use windows_core::HSTRING;
+
+// FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS
+bitflags::bitflags! {
+    #[derive(Debug, Clone)]
+    pub struct ServiceNotificationFilterFlags: i32{
+        const None = FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS_NONE.0;
+        const NamePrefix = FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS_NAME_PREFIX.0;
+        const PrimaryOnly = FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS_PRIMARY_ONLY.0;
+    }
+}
+
+impl From<&ServiceNotificationFilterFlags> for FABRIC_SERVICE_NOTIFICATION_FILTER_FLAGS {
+    fn from(value: &ServiceNotificationFilterFlags) -> Self {
+        Self(value.bits())
+    }
+}
+
+// FABRIC_SERVICE_NOTIFICATION_FILTER_DESCRIPTION
+#[derive(Debug, Clone)]
+pub struct ServiceNotificationFilterDescription {
+    pub name: HSTRING,
+    pub flags: ServiceNotificationFilterFlags,
+}
+
+impl From<&ServiceNotificationFilterDescription>
+    for FABRIC_SERVICE_NOTIFICATION_FILTER_DESCRIPTION
+{
+    fn from(value: &ServiceNotificationFilterDescription) -> Self {
+        Self {
+            Name: FABRIC_URI(value.name.as_ptr() as *mut u16),
```

Comment:
Does FABRIC_URI internally perform a deep copy on value pointed to by the pointer passed as a parameter?

Reply:
It is not the deep copy. The SF raw type returned from this conversion should have the same lifetime as the original value. I will add a note in the next PR.